commit d4fa4600673336322070c5751fcb79213b49f123
Author: Lakshmi Shanmugam <lshanmugam>
Date:   Mon Aug 9 13:14:30 2010 +0000

    Bug 314622 - Modify examples to use Display.getSystemCursor() instead of new Cursor()/dispose()

2	5	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java
2	8	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java
2	5	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesSWTTab.java
2	5	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesTab.java
9	18	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java
index d16cdfe..0cdb886 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/addressbook/AddressBook.java
@@ -201,7 +201,7 @@ private String encodeLine(String[] tableItems) {
 	return line;
 }
 private boolean findEntry() {
-	Cursor waitCursor = new Cursor(shell.getDisplay(), SWT.CURSOR_WAIT);
+	Cursor waitCursor = shell.getDisplay().getSystemCursor(SWT.CURSOR_WAIT);
 	shell.setCursor(waitCursor);
 	
 	boolean matchCase = searchDialog.getMatchCase();
@@ -229,7 +229,6 @@ private boolean findEntry() {
 	}
 	
 	shell.setCursor(null);
-	waitCursor.dispose();
 		
 	return found;
 }
@@ -279,7 +278,7 @@ private void openAddressBook() {
 		return;
 	}
 	
-	Cursor waitCursor = new Cursor(shell.getDisplay(), SWT.CURSOR_WAIT);
+	Cursor waitCursor = shell.getDisplay().getSystemCursor(SWT.CURSOR_WAIT);
 	shell.setCursor(waitCursor);
 	
 	FileReader fileReader = null;
@@ -305,7 +304,6 @@ private void openAddressBook() {
 	} finally {	
 		
 		shell.setCursor(null);
-		waitCursor.dispose();
 	
 		if(fileReader != null) {
 			try {
@@ -368,7 +366,6 @@ private boolean save() {
 		return false;
 	} finally {
 		shell.setCursor(null);
-		waitCursor.dispose();
 		
 		if(fileWriter != null) {
 			try {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java
index 914dc96..11b601c 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/fileviewer/IconCache.java
@@ -92,7 +92,7 @@ class IconCache {
 		if (stockCursors == null) {
 			stockCursors = new Cursor[] {
 				null,
-				new Cursor(display, SWT.CURSOR_WAIT)
+				display.getSystemCursor(SWT.CURSOR_WAIT)
 			};
 		}
 		iconCache = new Hashtable();
@@ -114,13 +114,7 @@ class IconCache {
 				image.dispose();
 			}
 		}
-		if (stockCursors != null) {
-			for (int i = 0; i < stockCursors.length; ++i) {
-				final Cursor cursor = stockCursors[i];
-				if (cursor != null) cursor.dispose();
-			}
-			stockCursors = null;
-		}
+		stockCursors = null;
 	}
 	/**
 	 * Creates a stock image
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesSWTTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesSWTTab.java
index 156e7aa..3924666 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesSWTTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesSWTTab.java
@@ -91,10 +91,7 @@ public void dispose() {
 	if (mouseMoveListener != null)
 		example.canvas.removeMouseMoveListener(mouseMoveListener);
 	
-	if (cursor != null) {
-		cursor.dispose();
-		cursor = null;
-	}
+	cursor = null;
 }
 
 	/**
@@ -125,7 +122,7 @@ private boolean isHovering(MouseEvent e) {
  */
 public void createControlPanel(Composite parent) {
 	if (cursor == null) { 
-		cursor = new Cursor(parent.getDisplay(), SWT.CURSOR_HAND);
+		cursor = parent.getDisplay().getSystemCursor(SWT.CURSOR_HAND);
 	}
 
 	mouseMoveListener = new MouseMoveListener() {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesTab.java
index 9784154..a621168 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/graphics/CurvesTab.java
@@ -80,10 +80,7 @@ public void dispose() {
 	if (mouseMoveListener != null)
 		example.canvas.removeMouseMoveListener(mouseMoveListener);
 	
-	if (cursor != null) {
-		cursor.dispose();
-		cursor = null;
-	}
+	cursor = null;
 }
 
 /**
@@ -110,7 +107,7 @@ private boolean isHovering(MouseEvent e) {
  */
 public void createControlPanel(Composite parent) {
 	if (cursor == null) { 
-		cursor = new Cursor(parent.getDisplay(), SWT.CURSOR_HAND);
+		cursor = parent.getDisplay().getSystemCursor(SWT.CURSOR_HAND);
 	}
 
 	mouseMoveListener = new MouseMoveListener() {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
index ccb0894..88935d0 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/imageanalyzer/ImageAnalyzer.java
@@ -193,7 +193,6 @@ public class ImageAnalyzer {
 				greenColor.dispose();
 				blueColor.dispose();
 				fixedWidthFont.dispose();
-				crossCursor.dispose();
 			}
 		});
 
@@ -204,7 +203,7 @@ public class ImageAnalyzer {
 		greenColor = new Color(display, 0, 255, 0);
 		blueColor = new Color(display, 0, 0, 255);
 		fixedWidthFont = new Font(display, "courier", 10, 0);
-		crossCursor = new Cursor(display, SWT.CURSOR_CROSS);
+		crossCursor = display.getSystemCursor(SWT.CURSOR_CROSS);
 		
 		// Add a menu bar and widgets.
 		createMenuBar();
@@ -768,7 +767,7 @@ public class ImageAnalyzer {
 	void menuComposeAlpha(int alpha_op) {
 		if (image == null) return;
 		animate = false; // stop any animation in progress
-		Cursor waitCursor = new Cursor(display, SWT.CURSOR_WAIT);
+		Cursor waitCursor = display.getSystemCursor(SWT.CURSOR_WAIT);
 		shell.setCursor(waitCursor);
 		imageCanvas.setCursor(waitCursor);
 		try {
@@ -798,7 +797,6 @@ public class ImageAnalyzer {
 		} finally {
 			shell.setCursor(null);
 			imageCanvas.setCursor(crossCursor);
-			waitCursor.dispose();
 		}
 	}
 
@@ -817,7 +815,7 @@ public class ImageAnalyzer {
 		if (filename == null)
 			return;
 
-		Cursor waitCursor = new Cursor(display, SWT.CURSOR_WAIT);
+		Cursor waitCursor = display.getSystemCursor(SWT.CURSOR_WAIT);
 		shell.setCursor(waitCursor);
 		imageCanvas.setCursor(waitCursor);
 		try {
@@ -848,7 +846,6 @@ public class ImageAnalyzer {
 		} finally {
 			shell.setCursor(null);
 			imageCanvas.setCursor(crossCursor);
-			waitCursor.dispose();
 		}
 	}
 	
@@ -866,7 +863,7 @@ public class ImageAnalyzer {
 		if (filename == null)
 			return;
 
-		Cursor waitCursor = new Cursor(display, SWT.CURSOR_WAIT);
+		Cursor waitCursor = display.getSystemCursor(SWT.CURSOR_WAIT);
 		shell.setCursor(waitCursor);
 		imageCanvas.setCursor(waitCursor);
 		ImageLoader oldLoader = loader;
@@ -912,7 +909,6 @@ public class ImageAnalyzer {
 		} finally {
 			shell.setCursor(null);
 			imageCanvas.setCursor(crossCursor);
-			waitCursor.dispose();
 		}
 	}
 	
@@ -926,7 +922,7 @@ public class ImageAnalyzer {
 		String urlname = textPrompter.open();
 		if (urlname == null) return;
 
-		Cursor waitCursor = new Cursor(display, SWT.CURSOR_WAIT);
+		Cursor waitCursor = display.getSystemCursor(SWT.CURSOR_WAIT);
 		shell.setCursor(waitCursor);
 		imageCanvas.setCursor(waitCursor);
 		ImageLoader oldLoader = loader;
@@ -971,7 +967,6 @@ public class ImageAnalyzer {
 		} finally {
 			shell.setCursor(null);
 			imageCanvas.setCursor(crossCursor);
-			waitCursor.dispose();
 		}
 	}
 
@@ -1039,7 +1034,7 @@ public class ImageAnalyzer {
 			return;
 		}
 
-		Cursor waitCursor = new Cursor(display, SWT.CURSOR_WAIT);
+		Cursor waitCursor = display.getSystemCursor(SWT.CURSOR_WAIT);
 		shell.setCursor(waitCursor);
 		imageCanvas.setCursor(waitCursor);
 		try {
@@ -1054,7 +1049,6 @@ public class ImageAnalyzer {
 		} finally {
 			shell.setCursor(null);
 			imageCanvas.setCursor(crossCursor);
-			waitCursor.dispose();
 		}
 	}
 
@@ -1103,7 +1097,7 @@ public class ImageAnalyzer {
 				return;
 		}
 		
-		Cursor waitCursor = new Cursor(display, SWT.CURSOR_WAIT);
+		Cursor waitCursor = display.getSystemCursor(SWT.CURSOR_WAIT);
 		shell.setCursor(waitCursor);
 		imageCanvas.setCursor(waitCursor);
 		try {
@@ -1144,7 +1138,6 @@ public class ImageAnalyzer {
 		} finally {
 			shell.setCursor(null);
 			imageCanvas.setCursor(crossCursor);
-			waitCursor.dispose();
 		}
 	}
 
@@ -1187,7 +1180,7 @@ public class ImageAnalyzer {
 				return;
 		}
 		
-		Cursor waitCursor = new Cursor(display, SWT.CURSOR_WAIT);
+		Cursor waitCursor = display.getSystemCursor(SWT.CURSOR_WAIT);
 		shell.setCursor(waitCursor);
 		imageCanvas.setCursor(waitCursor);
 		try {
@@ -1203,7 +1196,6 @@ public class ImageAnalyzer {
 		} finally {
 			shell.setCursor(null);
 			imageCanvas.setCursor(crossCursor);
-			waitCursor.dispose();
 		}
 	}
 
@@ -1260,7 +1252,7 @@ public class ImageAnalyzer {
 	void menuReopen() {
 		if (currentName == null) return;
 		animate = false; // stop any animation in progress
-		Cursor waitCursor = new Cursor(display, SWT.CURSOR_WAIT);
+		Cursor waitCursor = display.getSystemCursor(SWT.CURSOR_WAIT);
 		shell.setCursor(waitCursor);
 		imageCanvas.setCursor(waitCursor);
 		try {
@@ -1288,7 +1280,6 @@ public class ImageAnalyzer {
 		} finally {	
 			shell.setCursor(null);
 			imageCanvas.setCursor(crossCursor);
-			waitCursor.dispose();
 		}
 	}
 	
